### ISSUE TITLE
How to regenerate the cache folder for the covers

### ISSUE THREAD

**Issue by yongbi85** [evilhero/mylar#2040]
Hello,

I’m running mylar on synology for quite some time and recently upgraded to dsm 6.1 which resulted in having to uninstall and reinstall mylar.
I do a backup of the config.ini and the db every day so i could restore them, however i didn’t think about backuping the cache folder so now all my comics have no cover.
Is there any way to restore them from the cover.jpg files present in each folder ?

**Comment by evilhero** [comment]
Unfortunately there isn't at this time at least.

If you leave Mylar running, it will refresh each series in your watchlist every 5 minutes or so (roughly) until they're all updated over the course of a given week. So as long as you don't have thousands of series in your watchlist it should get thru most of  your series after a day or so I'd imagine.

It might be a nice option to allow for cover.jpg = cache/series.jpg, but it would require abit of coding/testing although I do see the benefit in trying to do this probably under the maintenance banner (since it's not something that would be done very often, if even more than once).

I'll mark this as an enhancement, cause I'll come back to it at some point when there's some free time and try to knock it off...

**Comment by yongbi85** [comment]
Thanks for your answer, i asked because even refreshing the serie isnt regenerating the jpg in the cache folder

**Comment by evilhero** [comment]
Make sure you the option enabled in the config.ini ```Place cover.jpg into Comic Directory for each comic``` and that should allow for regeneration of the cover. If it's not, enable debug/verbose logging and try to refresh a series and then post the debug log in here so we can see where it might be tripping up (could be permissions, or an actual code problem)

**Comment by yongbi85** [comment]
Still no luck with getting covers


Timestamp	Level	Message
2018-08-04 20:53:59	INFO	Update complete
2018-08-04 20:53:59	DEBUG	ReScheduled job: DB Updater to 2018-08-04 18:58:59
2018-08-04 20:53:59	INFO	[FILE-RESCAN] I have physically found 1 issues, ignored 0 issues, snatched 0 issues, and accounted for 0 in an Archived state [ Total Issue Count: 1 / 1 ]
2018-08-04 20:53:59	DEBUG	Successfully changed ownership and permissions [sc-mylar:mylar] / [0777 / 0777]
2018-08-04 20:53:59	DEBUG	[FILE-RESCAN] Ensuring permissions/ownership enforced for series: 100th Anniversary Special: Avengers
2018-08-04 20:53:59	INFO	[FILE-RESCAN] Total files located: 1
2018-08-04 20:53:58	DEBUG	[FILE-RESCAN] .......filesize: 70838567
2018-08-04 20:53:58	DEBUG	[FILE-RESCAN] .......filename: 100th Anniversary Special Avengers 001 (2014).cbz
2018-08-04 20:53:58	DEBUG	[FILE-RESCAN] Matched...issue: 100th Anniversary Special: Avengers#1 --- 1000
2018-08-04 20:53:58	DEBUG	[FILE-RESCAN] [460373] Issue match - fcdigit: 1000 ... int_iss: 1000
2018-08-04 20:53:58	DEBUG	[FILE-RESCAN] temploc: 001
2018-08-04 20:53:58	DEBUG	[ANNUAL-CHK] No annuals with identical issue numbering across annual volumes were detected for this series
2018-08-04 20:53:58	DEBUG	No issues with identical issue numbering were detected for this series
2018-08-04 20:53:58	DEBUG	series title possibly: 100 th Anniversary Special Avengers
2018-08-04 20:53:58	DEBUG	sf_highest_series_pos: ['100', 'th', 'Anniversary', 'Special', 'Avengers']
2018-08-04 20:53:58	DEBUG	issue verified as : 001
2018-08-04 20:53:58	DEBUG	issue_position: 5
2018-08-04 20:53:58	DEBUG	possible_issuenumbers: [{'position': 5, 'validcountchk': False, 'number': '001', 'mod_position': 35}]
2018-08-04 20:53:58	DEBUG	highest_series_position: 6
2018-08-04 20:53:58	DEBUG	year verified as: 2014
2018-08-04 20:53:58	DEBUG	year verified as : 2014
2018-08-04 20:53:58	DEBUG	sctd: 2018
2018-08-04 20:53:58	DEBUG	datecheck: [{'date': '2014', 'position': 6, 'mod_position': 39}]
2018-08-04 20:53:58	DEBUG	date: 2014
2018-08-04 20:53:58	DEBUG	checking date : 2014
Timestamp	Level	Message
2018-08-04 20:53:58	DEBUG	[FILENAME]: 100th Anniversary Special Avengers 001 (2014).cbz
2018-08-04 20:53:58	INFO	there are 1 files.
2018-08-04 20:53:58	INFO	[FILE-RESCAN] Now checking files for 100th Anniversary Special: Avengers (2014) in /volume1/Comics/Marvel/100th Anniversary Special Avengers/V(2014)
2018-08-04 20:53:58	INFO	I have added 0 new issues for this series that were not present before.
2018-08-04 20:53:58	INFO	In the process of converting the data to CV, I changed the status of 1 issues.
2018-08-04 20:53:58	DEBUG	Attempting to put the Status' back how they were.
2018-08-04 20:53:57	DEBUG	Deleting all old issue data to make sure new data is clean...
2018-08-04 20:53:57	INFO	returning to dbupdate module
2018-08-04 20:53:57	DEBUG	no update required - lastissueid [None] = latestissueid [460373]
2018-08-04 20:53:57	DEBUG	initiating issue updating - just the info
2018-08-04 20:53:57	INFO	Now adding/updating issues for 100th Anniversary Special: Avengers
2018-08-04 20:53:57	DEBUG	Finished Annual checking.
2018-08-04 20:53:57	DEBUG	there are 0 search results...
2018-08-04 20:53:53	DEBUG	[IMPORTER-ANNUAL] - No results, removing the year from the agenda and re-querying.
2018-08-04 20:53:53	DEBUG	there are 0 search results...
2018-08-04 20:53:50	DEBUG	[IMPORTER-ANNUAL] - Annual Year:2014
2018-08-04 20:53:50	DEBUG	checking annual db
2018-08-04 20:53:50	DEBUG	issuetype: None
2018-08-04 20:53:50	DEBUG	latestissueinfo: [{'latestiss': u'1', 'latestdate': u'2014-09-30'}]
2018-08-04 20:53:50	DEBUG	issuechk: None
2018-08-04 20:53:50	DEBUG	calledfrom: dbupdate
2018-08-04 20:53:50	DEBUG	comicissues: 1
2018-08-04 20:53:50	DEBUG	comicname: 100th Anniversary Special: Avengers
2018-08-04 20:53:50	DEBUG	comicid: 75888
2018-08-04 20:53:50	DEBUG	issuedata call references...

Timestamp	Level	Message
2018-08-04 20:53:50	INFO	Sucessfully retrieved issue details for 100th Anniversary Special: Avengers
2018-08-04 20:53:50	DEBUG	querying range from 0 to 100
2018-08-04 20:53:50	DEBUG	there are 1 search results...
2018-08-04 20:53:46	DEBUG	Cover already exists for series. Not redownloading.
2018-08-04 20:53:46	INFO	Directory (/volume1/Comics/Marvel/100th Anniversary Special Avengers/V(2014)) already exists! Continuing...
2018-08-04 20:53:46	INFO	Previous version detected as None - seeing if update required
2018-08-04 20:53:46	INFO	Sucessfully retrieved details for 100th Anniversary Special: Avengers
2018-08-04 20:53:46	INFO	Now adding/updating: 100th Anniversary Special: Avengers
2018-08-04 20:53:46	DEBUG	comic[ComicVersion]:None
2018-08-04 20:53:46	DEBUG	vari is set to: no
2018-08-04 20:53:46	DEBUG	number of issues CV says it has: 1
2018-08-04 20:53:46	DEBUG	number of issues I counted: 1
2018-08-04 20:53:43	INFO	[DIRECTORY-CHECK] Found comic directory: /volume1/Comics/Marvel/100th Anniversary Special Avengers/V(2014)
2018-08-04 20:53:43	DEBUG	Refreshing the series and pulling in new data using only CV.
2018-08-04 20:53:43	DEBUG	checking annual db
2018-08-04 20:53:43	DEBUG	Checking how out of whack the series is.
2018-08-04 20:53:43	DEBUG	Gathering the status of all issues for the series.
2018-08-04 20:53:43	DEBUG	CV_OneTimer option enabled...
2018-08-04 20:53:43	DEBUG	Refreshing: 100th Anniversary Special: Avengers (2014) [75888]
2018-08-04 20:53:43	DEBUG	Refreshing comic: [u'75888']
